source/MapPanel.cpp

```diff
@@ -1152,17 +1161,38 @@ void MapPanel::DrawMissions()
 	const Color &blockedColor = *colors.Get("blocked mission");
 	const Color &specialColor = *colors.Get("special mission");
 	const Color &waypointColor = *colors.Get("waypoint");
+	if(specialSystem)
+	{
+		// The special system pointer is larger than the others.
+		++pointerCount[specialSystem].drawn;
+		Angle a = Angle(30. * pointerCount[specialSystem].drawn);
+		Point pos = Zoom() * (specialSystem->Position() + center);
+		PointerShader::Draw(pos, a.Unit(), 20.f, 27.f, -4.f, black);
+		PointerShader::Draw(pos, a.Unit(), 11.5f, 21.5f, -6.f, specialColor);
+	}
+	// Calculate the available (and unavailable) jobs, but don't draw them yet.
 	for(const Mission &mission : player.AvailableJobs())
 	{
 		const System *system = mission.Destination()->GetSystem();
-		DrawPointer(system, angle[system], mission.CanAccept(player) ? availableColor : unavailableColor);
+		auto &&it = pointerCount[system];
+		if(mission.CanAccept(player))
+			++it.available;
+		else
+			++it.unavailable;
 	}
 	for(const Mission &mission : player.Missions())
 	{
 		if(!mission.IsVisible())
 			continue;
 
 		const System *system = mission.Destination()->GetSystem();
+
+		// Reserve a maximum of 6 slots for available missions.
+		auto &&it = pointerCount[system];
+		int reserved = min(6, it.available + it.unavailable);
+		if(it.drawn >= 12 - reserved)
```

Comment:
Make the 12 a global constant with a cool name (since it's used in two places). Maybe also for the `30.` (degrees), or we calculate the degree from the constant if that makes sense.